samples/12.customization-minimizable-web-chat/src/MinimizableWebChat.js

```diff
@@ -1,116 +1,112 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
 import { createStore, createStyleSet } from 'botframework-webchat';
 
 import WebChat from './WebChat';
 
 import './fabric-icons-inline.css';
 import './MinimizableWebChat.css';
 
-export default class extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.handleFetchToken = this.handleFetchToken.bind(this);
-    this.handleMaximizeButtonClick = this.handleMaximizeButtonClick.bind(this);
-    this.handleMinimizeButtonClick = this.handleMinimizeButtonClick.bind(this);
-    this.handleSwitchButtonClick = this.handleSwitchButtonClick.bind(this);
-
-    const store = createStore({}, ({ dispatch }) => next => action => {
-      if (action.type === 'DIRECT_LINE/CONNECT_FULFILLED') {
-        setTimeout(() => {
-          dispatch({
-            type: 'WEB_CHAT/SEND_EVENT',
-            payload: {
-              name: 'webchat/join',
-              value: {
-                language: window.navigator.language
+const MinimizableWebChat = () => {
+  const store = useMemo(
+    () =>
+      createStore({}, ({ dispatch }) => next => action => {
+        if (action.type === 'DIRECT_LINE/CONNECT_FULFILLED') {
+          setTimeout(() => {
```

Comment:
Why do we have this in a timeout?

Reply:
Yes, checked we don't need that. I was naively converting the code. 😆